### ISSUE TITLE
Can't pass in custom eval function

### ISSUE THREAD

**Issue by jgrimes** [Lambda-X/replumb#199]
While you can customize many things about how replumb loads/writes files, you are forced to use `js/eval` for evaluation, which precludes using things like Node's `vm.runInThisContext`.

Incoming PR adding basic support for this.

**Comment by arichiardi** [comment]
Oh ok I see now the point of adding it.
The thing is, given that we already support nodejs, this should probably supported upstream...if really necessary for successful execution in nodejs...I am not familiar of what the function does, can you summarize it briefly please?

**Comment by jgrimes** [comment]
v8 has multiple ways of evaluating javascript code, mostly to allow evaluating JS code in different contexts. The node.js docs describe a context as:

> In V8, a context is an execution environment that allows separate, unrelated, JavaScript applications to run in a single instance of V8. You must explicitly specify the context in which you want any JavaScript code to be run.

What I am trying to do is load ClojureScript files into a running Atom (nodejs based text editor) instance. Some forms of eval'ing javascript are disabled but there are workarounds for eval'ing javascript code --  _if_ the evaluation is synchronous -- that involve setting `global.eval` to `vm.runInThisContext`, evaling, and setting it back. 

This doesn't work with my code calling bootstrapped ClojureScript because it is all async, but if I can just supply the eval fn to use directly then it doesn't matter.

**Comment by arichiardi** [comment]
Understand now, so I am fine with the PR actually, let's maybe rename the key to `js-eval` to make it clear this is going to replace that part only, and then you need to add it to (https://github.com/Lambda-X/replumb/blob/master/test/cljs/replumb/options_test.cljs#L7)[this]
